packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -853,5 +874,139 @@ export const getAllVisualExperiments = async (
     visualChangeset: c,
   }));
 
-  return visualExperiments.filter(isValidVisualExperiment);
+  return visualExperiments.filter(_isValidVisualExperiment);
+};
+
+const _getPayloadKeys = (
+  organization: OrganizationInterface,
+  experiment: ExperimentInterface
+): SDKPayloadKey[] => {
+  const environments =
+    organization.settings?.environments?.map((e) => e.id) ?? [];
+  const project = experiment.project ?? "";
+  return environments.map((e) => ({
+    environment: e,
+    project,
+  }));
+};
+
+const _getAllVisualChanges = async (
+  organization: OrganizationInterface,
+  experiment: ExperimentInterface
+) => {
+  const visualChangesets = await findVisualChangesetsByExperiment(
+    experiment.id,
+    organization.id
+  );
+
+  if (!visualChangesets.length) return [];
+
+  return flatten(visualChangesets.map((vc) => vc.visualChanges));
+};
+
+const _hasVisualChanges = async (
+  organization: OrganizationInterface,
+  experiment: ExperimentInterface
+) => {
+  const visualChanges = await _getAllVisualChanges(organization, experiment);
+  return visualChanges.some((vc) => !!vc.css || !!vc.domMutations.length);
+};
+
+const _fireWebhoks = async (
+  experiment: ExperimentInterface,
+  previousProject: string = ""
+) => {
+  const payloadKeys = getSDKPayloadKeys(
+    new Set(["dev", "production"]),
+    new Set(["", previousProject || "", experiment.project || ""])
+  );
+
+  // fire the webhook:
+  await queueWebhook(experiment.organization, payloadKeys, false);
+
+  // invalidate the CDN
+  await queueCDNInvalidate(experiment.organization, (key) => {
+    // Which url to invalidate depends on the type of experiment
+    return experiment.implementation === "visual"
+      ? `/js/${key}.js`
+      : `/config/${key}`;
+  });
+};
```

Comment:
`_fireWebhooks` is not new - it's the old `experimentUpdated` function - code is untouched, just renamed under a new function name.